system-x/services/db/mongodb/src/main/java/software/tnb/db/mongodb/resource/openshift/OpenshiftMongoDB.java

```diff
@@ -150,7 +150,11 @@ protected MongoClient client() {
 
     @Override
     public String replicaSetUrl() {
-        return String.format("mongodb://%s:%s@%s:%d/%s", account().username(), account().password(), hostname(), localPort, account().database());
+        return replicaSetUrl(localPort);
+    }
+
+    public String replicaSetUrl(int port) {
+        return String.format("mongodb://%s:%s@%s:%d/%s", account().username(), account().password(), hostname(), port, account().database());
     }
```

Comment:
this change won't work, as the `replicaSetUrl` is used for two things:

- how the integration (running on openshift) connects to the mongodb
- how the client (running on your machine) connects to the mongodb (through portforward)

This change will "fix" it for the client, but the integration won't be able to connect to mongo, because it will try to connect to the `<portforward>` port.

I checked the code and the change should be:
- bring back `port()` instead of `localPort` in the `replicaSetUrl`
- do the same [replacement](https://github.com/tnb-software/TNB/blob/main/system-x/services/db/mongodb/src/main/java/software/tnb/db/mongodb/resource/openshift/OpenshiftMongoDB.java#L124) for port on that line, e.g. replace `:port()` for `:localPort`

Reply:
I see - thank you, will fix